src/main/java/dev/jbang/dependencies/ArtifactResolver.java

```diff
@@ -0,0 +1,459 @@
+														&& a.getArtifactId().equals(art.getArtifactId()))
+												.findFirst();
+			if (ma.isPresent()) {
+				return new Dependency(ma.get(), d.getScope(), d.getOptional(), d.getExclusions());
+			}
+		}
+		return d;
+	}
+
+	private List<Dependency> getManagedDependencies(Dependency dependency) {
+		return resolveDescriptor(dependency.getArtifact()).getManagedDependencies();
+	}
+
+	private ArtifactDescriptorResult resolveDescriptor(Artifact artifact) {
+		try {
+			ArtifactDescriptorRequest descriptorRequest = new ArtifactDescriptorRequest()
+																							.setArtifact(artifact)
+																							.setRepositories(
+																									repositories);
+			return system.readArtifactDescriptor(session, descriptorRequest);
+		} catch (ArtifactDescriptorException ex) {
+			throw new ExitException(1, "Could not read artifact descriptor for " + artifact, ex);
+		}
+	}
+
+	private RemoteRepository toRemoteRepo(MavenRepo repo) {
+		return new RemoteRepository.Builder(repo.getId(), "default", repo.getUrl())
+																					.setPolicy(getUpdatePolicy())
+																					.build();
+	}
+
+	private static Dependency toDependency(Artifact artifact) {
+		return new Dependency(artifact,
+				"pom".equalsIgnoreCase(artifact.getExtension()) ? "import" : JavaScopes.COMPILE);
+	}
+
+	private static Artifact toArtifact(String coord) {
+		return toArtifact(MavenCoordinate.fromString(coord));
+	}
+
+	private static Artifact toArtifact(MavenCoordinate coord) {
+		return new DefaultArtifact(coord.getGroupId(), coord.getArtifactId(), coord.getClassifier(),
+				coord.getPackaging(), coord.getVersion());
+	}
+
+	private static ArtifactInfo toArtifactInfo(Artifact artifact) {
+		MavenCoordinate coord = new MavenCoordinate(artifact.getGroupId(), artifact.getArtifactId(),
+				artifact.getVersion(), artifact.getClassifier(), artifact.getExtension());
+		return new ArtifactInfo(coord, artifact.getFile().toPath());
+	}
+
+	private RepositorySystem newRepositorySystem() {
+		DefaultServiceLocator locator = MavenRepositorySystemUtils	.newServiceLocator()
+																	.addService(RepositoryConnectorFactory.class,
+																			BasicRepositoryConnectorFactory.class)
+																	.addService(TransporterFactory.class,
+																			FileTransporterFactory.class)
+																	.addService(TransporterFactory.class,
+																			HttpTransporterFactory.class);
+		return locator.getService(RepositorySystem.class);
+	}
+
+	private Settings newEffectiveSettings() {
+		DefaultSettingsBuilderFactory factory = new DefaultSettingsBuilderFactory();
+		DefaultSettingsBuilder builder = factory.newInstance();
+
+		SettingsBuildingRequest settingsBuilderRequest = new DefaultSettingsBuildingRequest();
+		settingsBuilderRequest.setSystemProperties(System.getProperties());
+
+		// if we have a root folder, don't read settings at all
+		if (withUserSettings && (localFolderOverride == null || settingsXml != null)) {
```

Comment:
This if lead to my settings.xml not being read at all

Reply:
Indeed. I've simplified it and should now pass.